tools/wake-format/formatter.h

```diff
@@ -347,6 +347,85 @@ struct FmtPredicate<std::initializer_list<T>> {
   bool operator()(wcl::doc_builder& builder, ctx_t ctx, CSTElement& node) { return set[node.id()]; }
 };
 
+struct EpsilonCase {
+  ALWAYS_INLINE bool run(wcl::doc_builder& builder, ctx_t ctx, CSTElement& node) { return false; }
+};
+
+template <class Predicate, class FMT>
+struct PredicateCase {
+  Predicate predicate;
+  FMT formatter;
+
+  PredicateCase(Predicate predicate, FMT formatter) : predicate(predicate), formatter(formatter) {}
+
+  ALWAYS_INLINE bool run(wcl::doc_builder& builder, ctx_t ctx, CSTElement& node) {
+    if (!predicate(builder, ctx, node)) {
+      return false;
+    }
+    builder.append(formatter.compose(ctx.sub(builder), node));
+    return true;
+  }
+};
+
+template <class FMT>
+struct OtherwiseCase {
```

Comment:
I think anytime you use this you can instead just return the FMT underlying it. This is basically `IDFmt`

Reply:
We need the `return true` part